opensrp-app/src/main/java/org/smartregister/repository/LocationTagRepository.java

```diff
@@ -0,0 +1,141 @@
+            .registerTypeAdapter(LocationProperty.class, new PropertiesConverter()).create();
+
+    public static void createTable(SQLiteDatabase database) {
+        database.execSQL(CREATE_LOCATION_TAG_TABLE);
+        database.execSQL(CREATE_LOCATION_TAG_NAME_INDEX);
+    }
+
+    protected String getLocationTagTableName() {
+        return LOCATION_TAG_TABLE;
+    }
+
+    public void addOrUpdate(LocationTag locationTag) {
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(NAME, locationTag.getName());
+        contentValues.put(LOCATION_ID, locationTag.getLocationId());
+        getWritableDatabase().replace(getLocationTagTableName(), null, contentValues);
+
+    }
+
+    public List<LocationTag> getAllLocationTags() {
+        Cursor cursor = null;
+        List<LocationTag> locationsTags = new ArrayList<>();
+        try {
+            cursor = getReadableDatabase().rawQuery("SELECT * FROM " + getLocationTagTableName(), null);
+            while (cursor.moveToNext()) {
+                locationsTags.add(readCursor(cursor));
+            }
+            cursor.close();
+        } catch (Exception e) {
+            Timber.e(e);
+        } finally {
+            if (cursor != null)
+                cursor.close();
+        }
+        return locationsTags;
+    }
+
+    public List<LocationTag> getLocationTagByLocationId(String id) {
+        List<LocationTag> locationsTags = new ArrayList<>();
+        try (Cursor cursor = getReadableDatabase().rawQuery("SELECT * FROM " + getLocationTagTableName() +
+                " WHERE " + LOCATION_ID + " =?", new String[]{id})) {
+            while (cursor.moveToNext()) {
+                locationsTags.add(readCursor(cursor));
+            }
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+        return locationsTags;
+
+    }
+
+    /**
+     * Get a List of location tags that match the list of ids provided
+     *
+     * @param ids list of location ids
+     * @return the list of locations that match the params provided
+     */
+    public List<LocationTag> getLocationTagsByLocationIds(List<String> ids) {
+        return getLocationTagsByLocationIds(ids, true);
+    }
+
+    /**
+     * Get a List of location tags that either match or don't match the list of ids provided
+     * depending on value of the inclusive flag
+     *
+     * @param ids       list of location tag ids
+     * @param inclusive flag that determines whether the list of locations returned
+     *                  should include the locations whose ids match the params provided
+     *                  or exclude them
+     * @return
+     */
+    public List<LocationTag> getLocationTagsByLocationIds(List<String> ids, Boolean inclusive) {
+        Cursor cursor = null;
+        List<LocationTag> locationTags = new ArrayList<>();
+        int idCount = ids != null ? ids.size() : 0;
+        String[] idsArray = ids != null ? ids.toArray(new String[0]) : null;
```

Comment:
```suggestion
        String[] idsArray = ids != null ? ids.toArray(new String[ids.size()]) : null;
```

This is to make sure you use an array with same size as arraylist

Reply:
> This is to make sure you use an array with same size as arraylist
I have reverted the suggestion issued above. When converting lists to array it is suggested to use an empty array since jdk6 onwards to improve performance.